### ISSUE TITLE
Verification Failed for captain.caprover-root.mydomain.nu

### ISSUE THREAD

**Issue by Logstor** [caprover/caprover#1491]
### Problem
The current problem is that Captain and Nginx keep restarting, so I cannot access either apps or configuration files from the interface. It's only Captain and Nginx which is restarting not the different apps I've deployed with CapRover - they're running as they should.

The error is probably obvious from the logs below, but I can't seem to find what's wrong.

```
ID                          NAME                    IMAGE                      NODE          DESIRED STATE   CURRENT STATE               ERROR                       PORTS
ne69clcl84mwt4kphstdlwk5w   captain-captain.1       caprover/caprover:1.10.1   PappPrimary   Running         Running 5 seconds ago                                   *:3000->3000/tcp,*:3000->3000/tcp
yc0cfriou1dv83saq6atdvwl3    \_ captain-captain.1   caprover/caprover:1.10.1   PappPrimary   Shutdown        Failed 10 seconds ago       "task: non-zero exit (1)"   
bf7rtmf0hd22hpgv6r6crxkhr    \_ captain-captain.1   caprover/caprover:1.10.1   PappPrimary   Shutdown        Failed about a minute ago   "task: non-zero exit (1)"   
vfqlye7wslr11inn35ztgvcoy    \_ captain-captain.1   caprover/caprover:1.10.1   PappPrimary   Shutdown        Failed 3 minutes ago        "task: non-zero exit (1)"   
nx5onr70x83jenwoalxxzqphb    \_ captain-captain.1   caprover/caprover:1.10.1   PappPrimary   Shutdown        Failed 5 minutes ago        "task: non-zero exit (1)"   

```

#### Cause
It started right after I made some modifications to the NGINX Base Config File or the CapRover Config File - I don't remember anymore. A restart was initiated when Save & Update was clicked, I mistakenly refreshed the page, and then CapRover never came online again.

#### Things tried

1. Restarted my Hetzner server instance
2. Entered the nginx docker image, and checked the nginx.conf file which, to my surprise, was unaltered, and running `nginx -t` returns success.
3. Checked ufw (see below)

### Output from `ufw status`
```
Status: active

To                         Action      From
--                         ------      ----
80,443,996,2377,3000,4789,7946/tcp ALLOW       Anywhere                  
2377,4789,7946/udp         ALLOW       Anywhere                  
22/tcp                     ALLOW       Anywhere                  
80,443,996,2377,3000,4789,7946/tcp (v6) ALLOW       Anywhere (v6)             
2377,4789,7946/udp (v6)    ALLOW       Anywhere (v6)             
22/tcp (v6)                ALLOW       Anywhere (v6)  
``` 

### Output from `uname -a && lsb_release -a && free -h`

```
Linux PappPrimary 5.4.0-124-generic #140-Ubuntu SMP Thu Aug 4 02:23:37 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 20.04.4 LTS
Release:	20.04
Codename:	focal
              total        used        free      shared  buff/cache   available
Mem:          7.6Gi       1.3Gi       5.1Gi        12Mi       1.2Gi       6.0Gi
Swap:            0B          0B          0B

```

### Logs from continuously restarting nginx container
```
docker logs 79e38538c756 --follow
/docker-entrypoint.sh: /docker-entrypoint.d/ is not empty, will attempt to perform configuration
/docker-entrypoint.sh: Looking for shell scripts in /docker-entrypoint.d/
/docker-entrypoint.sh: Launching /docker-entrypoint.d/10-listen-on-ipv6-by-default.sh
10-listen-on-ipv6-by-default.sh: info: /etc/nginx/conf.d/default.conf is not a file or does not exist
/docker-entrypoint.sh: Launching /docker-entrypoint.d/20-envsubst-on-templates.sh
/docker-entrypoint.sh: Launching /docker-entrypoint.d/30-tune-worker-processes.sh
/docker-entrypoint.sh: Configuration complete; ready for start up
2022/08/15 19:29:23 [emerg] 1#1: duplicate location "/" in /etc/nginx/conf.d/captain.conf:223
nginx: [emerg] duplicate location "/" in /etc/nginx/conf.d/captain.conf:223

```

### Logs from `caprover/caprover:1.10.1` image

```
docker logs 2821545c2ae3 --follow
Captain Starting ...
August 15th 2022, 6:26:35.277 pm    Emptying generated and temp folders.
August 15th 2022, 6:26:35.284 pm    Ensuring directories are available on host. Started.
August 15th 2022, 6:26:35.292 pm    Ensuring directories are available on host. Finished.
August 15th 2022, 6:26:35.299 pm    Network captain-overlay-network is already attached to service: captain-captain
August 15th 2022, 6:26:35.307 pm    captain-captain (service) has already been connected to secret: captain-salt
August 15th 2022, 6:26:35.309 pm    Migration not needed, skipping.
August 15th 2022, 6:26:35.319 pm    Copying fake certificates...
August 15th 2022, 6:26:35.324 pm    Updating Load Balancer - Setting up NGINX conf file...
August 15th 2022, 6:26:35.325 pm    Locking NGINX configuration reloading...
August 15th 2022, 6:26:35.342 pm    SUCCESS: UNLocking NGINX configuration reloading...
August 15th 2022, 6:26:35.346 pm    Captain Nginx is already running.. 
August 15th 2022, 6:26:38.352 pm    Retrying to get NodeID for captain-nginx retry count:0
August 15th 2022, 6:26:41.359 pm    Retrying to get NodeID for captain-nginx retry count:1
August 15th 2022, 6:26:41.364 pm    Updating NGINX service...
August 15th 2022, 6:26:41.373 pm    Waiting for 5 seconds for nginx reload to take into effect
August 15th 2022, 6:26:46.374 pm    Pruning containers...
August 15th 2022, 6:26:46.375 pm    NGINX is fully set up and working...
August 15th 2022, 6:26:46.379 pm    Captain Certbot is already running.. 
August 15th 2022, 6:26:46.382 pm    Updating Certbot service...
August 15th 2022, 6:26:46.391 pm    executeCommand Container: captain-certbot certbot certificates --non-interactive 
August 15th 2022, 6:26:47.240 pm    **** Captain is initialized and ready to serve you! ****
August 15th 2022, 6:26:48.248 pm    Error: Wrapped: Verification Failed for captain.caprover-root.papp.nu
Error: Wrapped: Verification Failed for captain.caprover-root.papp.nu
    at errorize (/usr/src/app/built/utils/Logger.js:7:16)
    at Function.e (/usr/src/app/built/utils/Logger.js:27:21)
    at Request._callback (/usr/src/app/built/user/system/DomainResolveChecker.js:54:42)
    at self.callback (/usr/src/app/node_modules/request/request.js:185:22)
    at Request.emit (events.js:400:28)
    at Request.onRequestError (/usr/src/app/node_modules/request/request.js:877:8)
    at ClientRequest.emit (events.js:400:28)
    at Socket.socketErrorListener (_http_client.js:475:9)
    at Socket.emit (events.js:400:28)
    at emitErrorNT (internal/streams/destroy.js:106:8)
August 15th 2022, 6:26:48.249 pm    Error: Wrapped: Error        Error: connect ECONNREFUSED 168.119.178.1:80
Error: Wrapped: Error        Error: connect ECONNREFUSED 168.119.178.1:80
    at errorize (/usr/src/app/built/utils/Logger.js:7:16)
    at Function.e (/usr/src/app/built/utils/Logger.js:27:21)
    at Request._callback (/usr/src/app/built/user/system/DomainResolveChecker.js:55:42)
    at self.callback (/usr/src/app/node_modules/request/request.js:185:22)
    at Request.emit (events.js:400:28)
    at Request.onRequestError (/usr/src/app/node_modules/request/request.js:877:8)
    at ClientRequest.emit (events.js:400:28)
    at Socket.socketErrorListener (_http_client.js:475:9)
    at Socket.emit (events.js:400:28)
    at emitErrorNT (internal/streams/destroy.js:106:8)
August 15th 2022, 6:26:48.249 pm    Error: Wrapped: body         undefined
Error: Wrapped: body         undefined
    at errorize (/usr/src/app/built/utils/Logger.js:7:16)
    at Function.e (/usr/src/app/built/utils/Logger.js:27:21)
    at Request._callback (/usr/src/app/built/user/system/DomainResolveChecker.js:56:42)
    at self.callback (/usr/src/app/node_modules/request/request.js:185:22)
    at Request.emit (events.js:400:28)
    at Request.onRequestError (/usr/src/app/node_modules/request/request.js:877:8)
    at ClientRequest.emit (events.js:400:28)
    at Socket.socketErrorListener (_http_client.js:475:9)
    at Socket.emit (events.js:400:28)
    at emitErrorNT (internal/streams/destroy.js:106:8)
August 15th 2022, 6:26:48.250 pm    Error: Wrapped: randomUuid   4def1008-602c-4f13-870f-b81dd75ee648
Error: Wrapped: randomUuid   4def1008-602c-4f13-870f-b81dd75ee648
    at errorize (/usr/src/app/built/utils/Logger.js:7:16)
    at Function.e (/usr/src/app/built/utils/Logger.js:27:21)
    at Request._callback (/usr/src/app/built/user/system/DomainResolveChecker.js:57:42)
    at self.callback (/usr/src/app/node_modules/request/request.js:185:22)
    at Request.emit (events.js:400:28)
    at Request.onRequestError (/usr/src/app/node_modules/request/request.js:877:8)
    at ClientRequest.emit (events.js:400:28)
    at Socket.socketErrorListener (_http_client.js:475:9)
    at Socket.emit (events.js:400:28)
    at emitErrorNT (internal/streams/destroy.js:106:8)
August 15th 2022, 6:26:48.250 pm    Captain health check failed: #0 at captain.caprover-root.papp.nu
August 15th 2022, 6:26:48.250 pm    NGINX health check failed: #0
August 15th 2022, 6:26:51.391 pm    Pruning containers...
August 15th 2022, 6:27:00.247 pm    Retrying to get containerId for captain-certbot retry count:0
August 15th 2022, 6:27:03.250 pm    Retrying to get containerId for captain-certbot retry count:1
August 15th 2022, 6:27:03.253 pm    executeCommand Container: captain-certbot certbot certificates --non-interactive 
August 15th 2022, 6:27:04.336 pm    executeCommand Container: captain-certbot certbot renew --non-interactive 
August 15th 2022, 6:27:07.354 pm    Updating Load Balancer - renewAllCerts
August 15th 2022, 6:27:07.354 pm    Locking NGINX configuration reloading...
August 15th 2022, 6:27:07.367 pm    SUCCESS: UNLocking NGINX configuration reloading...
August 15th 2022, 6:27:07.367 pm    sendReloadSignal...
August 15th 2022, 6:27:09.259 pm    Error: Wrapped: Verification Failed for captain.caprover-root.papp.nu
Error: Wrapped: Verification Failed for captain.caprover-root.papp.nu
    at errorize (/usr/src/app/built/utils/Logger.js:7:16)
    at Function.e (/usr/src/app/built/utils/Logger.js:27:21)
    at Request._callback (/usr/src/app/built/user/system/DomainResolveChecker.js:54:42)
    at self.callback (/usr/src/app/node_modules/request/request.js:185:22)
    at Request.emit (events.js:400:28)
    at Request.onRequestError (/usr/src/app/node_modules/request/request.js:877:8)
    at ClientRequest.emit (events.js:400:28)
    at Socket.socketErrorListener (_http_client.js:475:9)
    at Socket.emit (events.js:400:28)
    at emitErrorNT (internal/streams/destroy.js:106:8)
August 15th 2022, 6:27:09.259 pm    Error: Wrapped: Error        Error: connect ECONNREFUSED 168.119.178.1:80
Error: Wrapped: Error        Error: connect ECONNREFUSED 168.119.178.1:80
    at errorize (/usr/src/app/built/utils/Logger.js:7:16)
    at Function.e (/usr/src/app/built/utils/Logger.js:27:21)
    at Request._callback (/usr/src/app/built/user/system/DomainResolveChecker.js:55:42)
    at self.callback (/usr/src/app/node_modules/request/request.js:185:22)
    at Request.emit (events.js:400:28)
    at Request.onRequestError (/usr/src/app/node_modules/request/request.js:877:8)
    at ClientRequest.emit (events.js:400:28)
    at Socket.socketErrorListener (_http_client.js:475:9)
    at Socket.emit (events.js:400:28)
    at emitErrorNT (internal/streams/destroy.js:106:8)
August 15th 2022, 6:27:09.259 pm    Error: Wrapped: body         undefined
Error: Wrapped: body         undefined
    at errorize (/usr/src/app/built/utils/Logger.js:7:16)
    at Function.e (/usr/src/app/built/utils/Logger.js:27:21)
    at Request._callback (/usr/src/app/built/user/system/DomainResolveChecker.js:56:42)
    at self.callback (/usr/src/app/node_modules/request/request.js:185:22)
    at Request.emit (events.js:400:28)
    at Request.onRequestError (/usr/src/app/node_modules/request/request.js:877:8)
    at ClientRequest.emit (events.js:400:28)
    at Socket.socketErrorListener (_http_client.js:475:9)
    at Socket.emit (events.js:400:28)
    at emitErrorNT (internal/streams/destroy.js:106:8)
August 15th 2022, 6:27:09.260 pm    Error: Wrapped: randomUuid   448af519-8c37-4fd7-8193-ee5fd38aaf40
Error: Wrapped: randomUuid   448af519-8c37-4fd7-8193-ee5fd38aaf40
    at errorize (/usr/src/app/built/utils/Logger.js:7:16)
    at Function.e (/usr/src/app/built/utils/Logger.js:27:21)
    at Request._callback (/usr/src/app/built/user/system/DomainResolveChecker.js:57:42)
    at self.callback (/usr/src/app/node_modules/request/request.js:185:22)
    at Request.emit (events.js:400:28)
    at Request.onRequestError (/usr/src/app/node_modules/request/request.js:877:8)
    at ClientRequest.emit (events.js:400:28)
    at Socket.socketErrorListener (_http_client.js:475:9)
    at Socket.emit (events.js:400:28)
    at emitErrorNT (internal/streams/destroy.js:106:8)
August 15th 2022, 6:27:09.260 pm    Captain health check failed: #1 at captain.caprover-root.papp.nu
August 15th 2022, 6:27:09.260 pm    NGINX health check failed: #1
August 15th 2022, 6:27:10.371 pm    Retrying to get containerId for captain-nginx retry count:0
August 15th 2022, 6:27:13.375 pm    Retrying to get containerId for captain-nginx retry count:1
August 15th 2022, 6:27:16.381 pm    Retrying to get containerId for captain-nginx retry count:2
August 15th 2022, 6:27:19.386 pm    Retrying to get containerId for captain-nginx retry count:3
August 15th 2022, 6:27:22.392 pm    Retrying to get containerId for captain-nginx retry count:4
August 15th 2022, 6:27:25.398 pm    Retrying to get containerId for captain-nginx retry count:5
August 15th 2022, 6:27:28.404 pm    Retrying to get containerId for captain-nginx retry count:6
August 15th 2022, 6:27:30.268 pm    Error: Wrapped: Verification Failed for captain.caprover-root.papp.nu
Error: Wrapped: Verification Failed for captain.caprover-root.papp.nu
    at errorize (/usr/src/app/built/utils/Logger.js:7:16)
    at Function.e (/usr/src/app/built/utils/Logger.js:27:21)
    at Request._callback (/usr/src/app/built/user/system/DomainResolveChecker.js:54:42)
    at self.callback (/usr/src/app/node_modules/request/request.js:185:22)
    at Request.emit (events.js:400:28)
    at Request.onRequestError (/usr/src/app/node_modules/request/request.js:877:8)
    at ClientRequest.emit (events.js:400:28)
    at Socket.socketErrorListener (_http_client.js:475:9)
    at Socket.emit (events.js:400:28)
    at emitErrorNT (internal/streams/destroy.js:106:8)
August 15th 2022, 6:27:30.268 pm    Error: Wrapped: Error        Error: connect ECONNREFUSED 168.119.178.1:80
Error: Wrapped: Error        Error: connect ECONNREFUSED 168.119.178.1:80
    at errorize (/usr/src/app/built/utils/Logger.js:7:16)
    at Function.e (/usr/src/app/built/utils/Logger.js:27:21)
    at Request._callback (/usr/src/app/built/user/system/DomainResolveChecker.js:55:42)
    at self.callback (/usr/src/app/node_modules/request/request.js:185:22)
    at Request.emit (events.js:400:28)
    at Request.onRequestError (/usr/src/app/node_modules/request/request.js:877:8)
    at ClientRequest.emit (events.js:400:28)
    at Socket.socketErrorListener (_http_client.js:475:9)
    at Socket.emit (events.js:400:28)
    at emitErrorNT (internal/streams/destroy.js:106:8)
August 15th 2022, 6:27:30.269 pm    Error: Wrapped: body         undefined
Error: Wrapped: body         undefined
    at errorize (/usr/src/app/built/utils/Logger.js:7:16)
    at Function.e (/usr/src/app/built/utils/Logger.js:27:21)
    at Request._callback (/usr/src/app/built/user/system/DomainResolveChecker.js:56:42)
    at self.callback (/usr/src/app/node_modules/request/request.js:185:22)
    at Request.emit (events.js:400:28)
    at Request.onRequestError (/usr/src/app/node_modules/request/request.js:877:8)
    at ClientRequest.emit (events.js:400:28)
    at Socket.socketErrorListener (_http_client.js:475:9)
    at Socket.emit (events.js:400:28)
    at emitErrorNT (internal/streams/destroy.js:106:8)
August 15th 2022, 6:27:30.270 pm    Error: Wrapped: randomUuid   2acf256b-1cd5-491e-988c-76c20ab4dcab
Error: Wrapped: randomUuid   2acf256b-1cd5-491e-988c-76c20ab4dcab
    at errorize (/usr/src/app/built/utils/Logger.js:7:16)
    at Function.e (/usr/src/app/built/utils/Logger.js:27:21)
    at Request._callback (/usr/src/app/built/user/system/DomainResolveChecker.js:57:42)
    at self.callback (/usr/src/app/node_modules/request/request.js:185:22)
    at Request.emit (events.js:400:28)
    at Request.onRequestError (/usr/src/app/node_modules/request/request.js:877:8)
    at ClientRequest.emit (events.js:400:28)
    at Socket.socketErrorListener (_http_client.js:475:9)
    at Socket.emit (events.js:400:28)
    at emitErrorNT (internal/streams/destroy.js:106:8)
August 15th 2022, 6:27:30.270 pm    Captain health check failed: #2 at captain.caprover-root.papp.nu
August 15th 2022, 6:27:30.271 pm    NGINX health check failed: #2
August 15th 2022, 6:27:31.408 pm    Retrying to get containerId for captain-nginx retry count:7
August 15th 2022, 6:27:34.414 pm    Retrying to get containerId for captain-nginx retry count:8
August 15th 2022, 6:27:37.420 pm    Retrying to get containerId for captain-nginx retry count:9
August 15th 2022, 6:27:37.424 pm    Error: No containerId is found
Error: No containerId is found
    at /usr/src/app/built/docker/DockerApi.js:683:19
    at processTicksAndRejections (internal/process/task_queues.js:95:5)
August 15th 2022, 6:27:51.279 pm    Error: Wrapped: Verification Failed for captain.caprover-root.papp.nu
Error: Wrapped: Verification Failed for captain.caprover-root.papp.nu
    at errorize (/usr/src/app/built/utils/Logger.js:7:16)
    at Function.e (/usr/src/app/built/utils/Logger.js:27:21)
    at Request._callback (/usr/src/app/built/user/system/DomainResolveChecker.js:54:42)
    at self.callback (/usr/src/app/node_modules/request/request.js:185:22)
    at Request.emit (events.js:400:28)
    at Request.onRequestError (/usr/src/app/node_modules/request/request.js:877:8)
    at ClientRequest.emit (events.js:400:28)
    at Socket.socketErrorListener (_http_client.js:475:9)
    at Socket.emit (events.js:400:28)
    at emitErrorNT (internal/streams/destroy.js:106:8)
August 15th 2022, 6:27:51.279 pm    Error: Wrapped: Error        Error: connect ECONNREFUSED 168.119.178.1:80
Error: Wrapped: Error        Error: connect ECONNREFUSED 168.119.178.1:80
    at errorize (/usr/src/app/built/utils/Logger.js:7:16)
    at Function.e (/usr/src/app/built/utils/Logger.js:27:21)
    at Request._callback (/usr/src/app/built/user/system/DomainResolveChecker.js:55:42)
    at self.callback (/usr/src/app/node_modules/request/request.js:185:22)
    at Request.emit (events.js:400:28)
    at Request.onRequestError (/usr/src/app/node_modules/request/request.js:877:8)
    at ClientRequest.emit (events.js:400:28)
    at Socket.socketErrorListener (_http_client.js:475:9)
    at Socket.emit (events.js:400:28)
    at emitErrorNT (internal/streams/destroy.js:106:8)
August 15th 2022, 6:27:51.279 pm    Error: Wrapped: body         undefined
Error: Wrapped: body         undefined
    at errorize (/usr/src/app/built/utils/Logger.js:7:16)
    at Function.e (/usr/src/app/built/utils/Logger.js:27:21)
    at Request._callback (/usr/src/app/built/user/system/DomainResolveChecker.js:56:42)
    at self.callback (/usr/src/app/node_modules/request/request.js:185:22)
    at Request.emit (events.js:400:28)
    at Request.onRequestError (/usr/src/app/node_modules/request/request.js:877:8)
    at ClientRequest.emit (events.js:400:28)
    at Socket.socketErrorListener (_http_client.js:475:9)
    at Socket.emit (events.js:400:28)
    at emitErrorNT (internal/streams/destroy.js:106:8)
August 15th 2022, 6:27:51.279 pm    Error: Wrapped: randomUuid   7461977b-9597-4949-b94a-179b351ceb44
Error: Wrapped: randomUuid   7461977b-9597-4949-b94a-179b351ceb44
    at errorize (/usr/src/app/built/utils/Logger.js:7:16)
    at Function.e (/usr/src/app/built/utils/Logger.js:27:21)
    at Request._callback (/usr/src/app/built/user/system/DomainResolveChecker.js:57:42)
    at self.callback (/usr/src/app/node_modules/request/request.js:185:22)
    at Request.emit (events.js:400:28)
    at Request.onRequestError (/usr/src/app/node_modules/request/request.js:877:8)
    at ClientRequest.emit (events.js:400:28)
    at Socket.socketErrorListener (_http_client.js:475:9)
    at Socket.emit (events.js:400:28)
    at emitErrorNT (internal/streams/destroy.js:106:8)
August 15th 2022, 6:27:51.280 pm    Captain health check failed: #3 at captain.caprover-root.papp.nu
August 15th 2022, 6:27:51.280 pm    NGINX health check failed: #3
August 15th 2022, 6:28:12.294 pm    Error: Wrapped: Verification Failed for captain.caprover-root.papp.nu
Error: Wrapped: Verification Failed for captain.caprover-root.papp.nu
    at errorize (/usr/src/app/built/utils/Logger.js:7:16)
    at Function.e (/usr/src/app/built/utils/Logger.js:27:21)
    at Request._callback (/usr/src/app/built/user/system/DomainResolveChecker.js:54:42)
    at self.callback (/usr/src/app/node_modules/request/request.js:185:22)
    at Request.emit (events.js:400:28)
    at Request.onRequestError (/usr/src/app/node_modules/request/request.js:877:8)
    at ClientRequest.emit (events.js:400:28)
    at Socket.socketErrorListener (_http_client.js:475:9)
    at Socket.emit (events.js:400:28)
    at emitErrorNT (internal/streams/destroy.js:106:8)
August 15th 2022, 6:28:12.294 pm    Error: Wrapped: Error        Error: connect ECONNREFUSED 168.119.178.1:80
Error: Wrapped: Error        Error: connect ECONNREFUSED 168.119.178.1:80
    at errorize (/usr/src/app/built/utils/Logger.js:7:16)
    at Function.e (/usr/src/app/built/utils/Logger.js:27:21)
    at Request._callback (/usr/src/app/built/user/system/DomainResolveChecker.js:55:42)
    at self.callback (/usr/src/app/node_modules/request/request.js:185:22)
    at Request.emit (events.js:400:28)
    at Request.onRequestError (/usr/src/app/node_modules/request/request.js:877:8)
    at ClientRequest.emit (events.js:400:28)
    at Socket.socketErrorListener (_http_client.js:475:9)
    at Socket.emit (events.js:400:28)
    at emitErrorNT (internal/streams/destroy.js:106:8)
August 15th 2022, 6:28:12.294 pm    Error: Wrapped: body         undefined
Error: Wrapped: body         undefined
    at errorize (/usr/src/app/built/utils/Logger.js:7:16)
    at Function.e (/usr/src/app/built/utils/Logger.js:27:21)
    at Request._callback (/usr/src/app/built/user/system/DomainResolveChecker.js:56:42)
    at self.callback (/usr/src/app/node_modules/request/request.js:185:22)
    at Request.emit (events.js:400:28)
    at Request.onRequestError (/usr/src/app/node_modules/request/request.js:877:8)
    at ClientRequest.emit (events.js:400:28)
    at Socket.socketErrorListener (_http_client.js:475:9)
    at Socket.emit (events.js:400:28)
    at emitErrorNT (internal/streams/destroy.js:106:8)
August 15th 2022, 6:28:12.295 pm    Captain health check failed: #4 at captain.caprover-root.papp.nu
August 15th 2022, 6:28:12.295 pm    NGINX health check failed: #4
August 15th 2022, 6:28:12.294 pm    Error: Wrapped: randomUuid   50abba0c-b111-469f-95c7-5f6d6d0a85b6
Error: Wrapped: randomUuid   50abba0c-b111-469f-95c7-5f6d6d0a85b6
    at errorize (/usr/src/app/built/utils/Logger.js:7:16)
    at Function.e (/usr/src/app/built/utils/Logger.js:27:21)
    at Request._callback (/usr/src/app/built/user/system/DomainResolveChecker.js:57:42)
    at self.callback (/usr/src/app/node_modules/request/request.js:185:22)
    at Request.emit (events.js:400:28)
    at Request.onRequestError (/usr/src/app/node_modules/request/request.js:877:8)
    at ClientRequest.emit (events.js:400:28)
    at Socket.socketErrorListener (_http_client.js:475:9)
    at Socket.emit (events.js:400:28)
    at emitErrorNT (internal/streams/destroy.js:106:8)
```

**Comment by matiasfacello** [comment]
Having algo the same problem.

Something I would like to add, I don't know if you have the same problem, is that my caprover setup failed when enabling SSL.

```
Setup CapRover machine on your server...

? have you already started CapRover container on your server? Yes
? IP address of your server: 92.119.129.86
? CapRover server root domain: rov.pristineshock.com
? new CapRover password (min 8 characters): [hidden]
? enter new CapRover password again: [hidden]
? "valid" email address to get certificate and enable HTTPS: ********@gmail
.com
✖ Enabling SSL... Takes a few seconds...

Error: StatusCodeError: 502 - "<html>\r\n<head><title>502 Bad Gateway</title>
</head>\r\n<body>\r\n<center><h1>502 Bad Gateway</h1></center>\r\n<hr>
<center>nginx</center>\r\n</body>\r\n</html>\r\n"

```


The Nginx is in constant crashing with the `task: non-zero exit (1)` error.

Tried with Ubuntu 18 and now trying in 20. Followed Docker docs and then Caprover docs for installing.

**Comment by githubsaturn** [comment]
@Logstor , see this: https://caprover.com/docs/troubleshooting.html#ive-made-a-change-to-the-nginx-config-that-broke-the-admin-ui

@matiasfacello  - your issue is different. When you get this error, can you please get logs from your nginx service?
```
docker service logs captain-nginx --since 5m
```

**Comment by matiasfacello** [comment]
Thanks for the reply!

Interesting. Thought the issues were pretty similar.


Here is the 
`docker service logs captain-nginx --since 5m`
```
captain-nginx.1.y7whj0w7kgyr@psserv    | /docker-entrypoint.sh: /docker-entrypoint.d/ is not empty, will attempt to perform configuration
captain-nginx.1.y7whj0w7kgyr@psserv    | /docker-entrypoint.sh: Looking for shell scripts in /docker-entrypoint.d/
captain-nginx.1.y7whj0w7kgyr@psserv    | /docker-entrypoint.sh: Launching /docker-entrypoint.d/10-listen-on-ipv6-by-default.sh
captain-nginx.1.y7whj0w7kgyr@psserv    | 10-listen-on-ipv6-by-default.sh: info: /etc/nginx/conf.d/default.conf is not a file or does not exist
captain-nginx.1.y7whj0w7kgyr@psserv    | /docker-entrypoint.sh: Launching /docker-entrypoint.d/20-envsubst-on-templates.sh
captain-nginx.1.y7whj0w7kgyr@psserv    | /docker-entrypoint.sh: Launching /docker-entrypoint.d/30-tune-worker-processes.sh
captain-nginx.1.y7whj0w7kgyr@psserv    | /docker-entrypoint.sh: Configuration complete; ready for start up
captain-nginx.1.y7whj0w7kgyr@psserv    | 2022/08/16 03:31:30 [error] 22#22: *1 connect() failed (113: No route to host) while connecting to upstream, client: 172.18.0.1, server: captain.rov.pristineshock.com, request: "GET /checkhealth HTTP/1.1", upstream: "http://10.0.1.2:3000/checkhealth", host: "captain.rov.pristineshock.com"
captain-nginx.1.y7whj0w7kgyr@psserv    | 2022/08/16 03:31:53 [error] 23#23: *5 connect() failed (113: No route to host) while connecting to upstream, client: 172.18.0.1, server: captain.rov.pristineshock.com, request: "GET /checkhealth HTTP/1.1", upstream: "http://10.0.1.2:3000/checkhealth", host: "captain.rov.pristineshock.com"
captain-nginx.1.y7whj0w7kgyr@psserv    | 2022/08/16 03:32:16 [error] 23#23: *9 connect() failed (113: No route to host) while connecting to upstream, client: 172.18.0.1, server: captain.rov.pristineshock.com, request: "GET /checkhealth HTTP/1.1", upstream: "http://10.0.1.2:3000/checkhealth", host: "captain.rov.pristineshock.com"
captain-nginx.1.y7whj0w7kgyr@psserv    | 2022/08/16 03:32:39 [error] 23#23: *12 connect() failed (113: No route to host) while connecting to upstream, client: 172.18.0.1, server: captain.rov.pristineshock.com, request: "GET /checkhealth HTTP/1.1", upstream: "http://10.0.1.2:3000/checkhealth", host: "captain.rov.pristineshock.com"
captain-nginx.1.y7whj0w7kgyr@psserv    | 2022/08/16 03:33:02 [error] 23#23: *15 connect() failed (113: No route to host) while connecting to upstream, client: 172.18.0.1, server: captain.rov.pristineshock.com, request: "GET /checkhealth HTTP/1.1", upstream: "http://10.0.1.2:3000/checkhealth", host: "captain.rov.pristineshock.com"

```

**Comment by githubsaturn** [comment]
Okay, your issue is completely separate. Please **open another issue** and enter this information:
- Are you using a VPS or a local machine? 
- Have you followed the standard installation as [mentioned here](https://caprover.com/docs/get-started.html#caprover-setup), or did you modify it in any forms? 
- Output of `uname -a && lsb_release -a && free -h`
- Output of `cat /captain/data/config-captain.json`
- Output of `docker ps -a`
- Output of `docker service ls`
- Output of `docker exec -it $(docker ps --filter name=captain-nginx -q) curl -v captain-captain:3000`

**Comment by Logstor** [comment]
> @Logstor , see this: https://caprover.com/docs/troubleshooting.html#ive-made-a-change-to-the-nginx-config-that-broke-the-admin-ui
> 
> @matiasfacello - your issue is different. When you get this error, can you please get logs from your nginx service?
> 
> ```
> docker service logs captain-nginx --since 5m
> ```

Thanks for the reply @githubsaturn !

I got the problem fixed. So it was an nginx configuration error in one of the apps that caused this problem. After that got reset, nginx stopped restarting, and I got everything running as normal again.

Can't believe I missed that on the documentation site, but it surely would have fixed problem too :muscle: 

Thanks again @githubsaturn. I will close this issue, and hope it will help others that doesn't read the docs thoroughly :rofl:
